### PR TITLE
TOV / Hydro-without-hydro / bhah_lib

### PR DESCRIPTION
**Pull request by leowerneck**
This pull requests introduces several hydrodynamics features to NRPy2, including:
  - A TOV reader & interpolator (for NRPy/TOVola initial data);
  - bhah_lib, to generate a library version of the BH@H code;
  - The hydro-without-hydro example; and
  - A start-to-finish example of how to generate bhah_lib.

### CODE REVIEW

File: nrpy/examples/hydro_without_hydro.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+)
+from nrpy.infrastructures.BHaH.MoLtimestepping import MoL
+
+par.set_parval_from_str("Infrastructure", "BHaH")
+
+# Code-generation-time parameters:
+project_name = "hydro_without_hydro"
+IDtype = "tov_interp"
+CoordSystem = "Spherical"
+LapseEvolutionOption = "OnePlusLog"
```
Comment (zachetienne) [inline nrpy/examples/hydro_without_hydro.py:41]: The gauge conditions are not consistent with what is needed for the hydro-without-hydro test.

File: nrpy/examples/hydro_without_hydro.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        project_dir=project_dir,
+        subdirectory="simd",
+    )
+Makefile.output_CFunctions_function_prototypes_and_construct_Makefile(
+    project_dir=project_dir,
+    project_name=project_name,
+    exec_or_library_name=project_name,
+    compiler_opt_option="default",
+)
+shutil.copyfile("tov_id/tov.asc", f"project/{project_name}/tov.asc")
```
Comment (zachetienne) [inline nrpy/examples/hydro_without_hydro.py:243]: Data files are not permitted in NRPy 2. Instead we need to port over the TOV solver from NRPy 1, and adjust this example file to generate the data file. Ideally we just use a simple gamma-law EOS.

File: nrpy/examples/tov_id/tov.asc
Comment (zachetienne) [inline nrpy/examples/tov_id/tov.asc:1]: Data files are not permitted in NRPy 2. Please see hydro_with_hydro.py comment.

File: nrpy/examples/hydro_without_hydro.py (hunk 2 above)
Comment (zachetienne) [reply nrpy/examples/hydro_without_hydro.py:243]: Update: TOVola has been imported into NRPy 2! Check out the example in `nrpy/examples/tovola_neutron_star.py`

Comment (zachetienne) [reply nrpy/examples/hydro_without_hydro.py:243]: It currently only supports simple polytropes (to sidestep GRHayL dependence for now), but hey, that's a start.

File: nrpy/examples/tov_id/tov.asc
Comment (zachetienne) [reply nrpy/examples/tov_id/tov.asc:1]: Update: TOVola has been imported into NRPy 2! Check out the example in `nrpy/examples/tovola_neutron_star.py`. It currently only supports simple polytropes (to sidestep GRHayL dependence for now), but hey, that's a start. Should be sufficient for this PR.

File: nrpy/examples/hydro_without_hydro.py (hunk 1 above)
Comment (leowerneck) [reply nrpy/examples/hydro_without_hydro.py]: Thank you for pointing this out; I didn't realize we used harmonic slicing for the hydro-without-hydro test. This has been fixed.

**Pull request merged**